eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -1259,39 +1266,77 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
+            // there were too many iterations, we stop negotiating and accept their fee
+            log.warning("could not agree on closing fees after {} iterations, accepting their closing fees ({})", MAX_NEGOTIATION_ITERATIONS, remoteClosingFee)
+            handleMutualClose(signedClosingTx, Left(d.copy(bestUnpublishedClosingTx_opt = Some(signedClosingTx)))) sending closingSignedRemoteFees
+          } else if (lastLocalClosingSigned_opt.flatMap(_.localClosingSigned.feeRange_opt).exists(r => r.min <= remoteClosingFee && remoteClosingFee <= r.max)) {
+            // they chose a fee inside our proposed fee range, so we close and send a closing_signed for that fee
+            handleMutualClose(signedClosingTx, Left(d.copy(bestUnpublishedClosingTx_opt = Some(signedClosingTx)))) sending closingSignedRemoteFees
+          } else if (d.commitments.localCommit.spec.toLocal == 0.msat) {
+            // we have nothing at stake so there is no need to negotiate, we accept their fee right away
+            handleMutualClose(signedClosingTx, Left(d.copy(bestUnpublishedClosingTx_opt = Some(signedClosingTx)))) sending closingSignedRemoteFees
           } else {
-            log.info("proposing closingFeeSatoshis={}", closingSigned.feeSatoshis)
-            val closingTxProposed1 = (d.closingTxProposed: @unchecked) match {
-              case previousNegotiations :+ currentNegotiation => previousNegotiations :+ (currentNegotiation :+ ClosingTxProposed(closingTx, closingSigned))
+            c.feeRange_opt match {
+              case Some(ClosingSignedTlv.FeeRange(minFee, maxFee)) if !d.commitments.localParams.isFunder =>
+                // if we are fundee and they proposed a fee range, we pick a value in that range and they should accept it without further negotiation
+                // we don't care much about the closing fee since they're paying it (not us) and we can use CPFP if we want to speed up confirmation
+                // but we need to ensure it's high enough to at least propagate across the network
+                val txPropagationFeerate = nodeParams.onChainFeeConf.feeEstimator.getFeeratePerKw(1008) * 2
+                val txPropagationMinFee = Transactions.weight2fee(txPropagationFeerate, signedClosingTx.tx.weight())
+                if (maxFee < txPropagationMinFee) {
+                  log.warning("their highest closing fee is below our tx propagation threshold (feerate={}): {} < {}", txPropagationFeerate, maxFee, txPropagationMinFee)
+                  stay() sending Warning(d.channelId, s"closing fee range must not be below $txPropagationMinFee ($txPropagationFeerate)")
+                } else {
+                  val closingFee = Closing.firstClosingFee(d.commitments, d.localShutdown.scriptPubKey, d.remoteShutdown.scriptPubKey, nodeParams.onChainFeeConf.feeEstimator, nodeParams.onChainFeeConf.feeTargets) match {
+                    case ClosingFees(preferred, _, _) if preferred > maxFee => maxFee
```

Comment:
See https://github.com/ACINQ/eclair/pull/1768#discussion_r694711406, I think it's dangerous to play with that limit

Reply:
I'm not saying that we should set our `min` equal or even close to the min propagation fee, but that we should make it always greater (possibly much greater). Satisfying our `min` should always satisfy the min propagation fee, this would simplify the logic.

Reply:
Right, understood, that makes sense indeed.

Reply:
Actually it shouldn't use our `minFee`, because we're fundee in that case so we should accept almost whatever they send us as long as it reasonably relays. Our `minFee` may be too restrictive here, so I chose to use the 1008 block target (which will always be greater than the minimum propagation fee).

I agree that it makes the logic a bit more complex, but I think it's useful.
We first check whether their fee is agreeable (the `txPropagationMinFee` condition) then we choose our preferred fee based on their fee range.
Those are two separate steps.

Reply:
> Actually it shouldn't use our minFee, because we're fundee in that case so we should accept almost whatever they send us as long as it reasonably relays. Our minFee may be too restrictive here

Our `minFee` should take into account whether we are funder or fundee. If we accept it, then it satisfies `minFee`, otherwise what is the meaning of `minFee`?

Reply:
I'll give it a try, I'm afraid it will require more code and be more complex but I agree that conceptually it's what makes sense.

Reply:
I would simply have added an additional constraint here:

https://github.com/ACINQ/eclair/blob/closing-fee-range/eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala#L457-467

```scala
val min = Math.max(preferredFeerate / 2, feeEstimator.getFeeratePerKw(1008))
```

Am I missing something?

Reply:
I did that in https://github.com/ACINQ/eclair/pull/1768/commits/2de7d7fbfcdda80c994981793f67a186110ba759, but we actually want `min`, not `max`.

The reason I thought this was complex was that I wanted to enforce this `min` condition on all instances of `ClosingFeerates` (and reject `close` RPC calls that set a minimum fee too low). But in fact this isn't doable, because the feerate may change between the time the user issues the command and `shutdown` starts and the time the tx is actually broadcast...